pkg/serverless/trace/inferredspan/span_enrichment_test.go

```diff
@@ -646,6 +646,7 @@ func TestEnrichInferredSpanWithEventBridgeEvent(t *testing.T) {
 	assert.Equal(t, "web", span.Type)
 	assert.Equal(t, "aws.eventbridge", span.Meta[operationName])
 	assert.Equal(t, "eventbridge.custom.event.sender", span.Meta[resourceNames])
+	assert.Equal(t, "testdetail", span.Meta[detailType])
```

Comment:
The `DetailType` field has a non-default json marshalling key `detail-type`. This test was added because it fails if this non-default json marshalling key is missing.

Reply:
Goofy JSON serializing is the worst thing about Smithy-generated libraries IMO